script/build.ts

```diff
@@ -171,7 +188,15 @@ function packageApp() {
     appBundleId: getBundleID(),
     appCategoryType: 'public.app-category.developer-tools',
     darwinDarkModeSupport: true,
-    osxSign: true,
+    osxSign: {
+      // @ts-ignore (the library's type definition is wrong)
```

Comment:
This is a heavy hammer we're using here, if I'm not mistaken it's ignoring this entire subtype (as opposed to just the subsequent line).

This makes me a little nervous. Is there any chance we could make what we're overriding be explicit? Looking at the type it seems what's missing is the `hardenedRuntime` and the `type` properties. Could we extend the custom type we already have to include these? If we add the following to the `IPackageAdditionalOptions` type...

```ts
readonly osxSign: packager.ElectronOsXSignOptions & {
  readonly hardenedRuntime?:boolean
  readonly type?: string
}
```

... combined with my comment below about removing the `identity` property we should be okay?

Reply:
ah thanks, i missed our existing type override. the `type` field wasn't a problem, so i just added `hardenedRuntime`. (yesterday i also opened https://github.com/electron/electron-osx-sign/pull/216 for fix things upstream.) 

(also from what i've read, [`@ts-ignore` only ever ignores errors for the one line after it](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#suppress-errors-in-ts-files-using--ts-ignore-comments), but i agree its a heavy hammer when we can easily override the type.)

https://github.com/desktop/desktop/pull/8555/commits/3b8675f5cbde5c0375e0c4e5d8971713eab8f421

Reply:
> also from what i've read, @ts-ignore only ever ignores errors for the one line after it, but i agree its a heavy hammer when we can easily override the type

This is what I thought as well but when I tried removing the comment as well as the `hardenedRuntime` property I still got an error for the `identity` type mismatch. 🤷‍♂